components/button/__test__/button.spec.ts

```diff
@@ -1,24 +1,49 @@
-import { render } from "@testing-library/svelte";
+import { render, fireEvent } from "@testing-library/svelte";
 import Button from "../src/Button.svelte";
+import SlotTest from "../../SlotTest/SlotTest.svelte";
 
 describe("Button", () => {
   const props = {
     class: "custom-class",
     name: "test-button",
     label: "Hello World!",
     style: "width: 100px;",
+    ref: document.createElement('button') as HTMLButtonElement
   };
 
-  const results = render(Button, { props });
+  it("should render correctly", () => {
+    const result = render(Button, { props });
+    expect(() => result).not.toThrow();
+  });
+
+  it("should render slots correctly", () => {
+    const {getByTestId} = render(SlotTest, {
+      props: { Component: Button, props }
+    });
+    expect(() => getByTestId("slot")).not.toThrow();
+  });
 
-  test("shows proper heading when rendered", () => {
-    const button = results.getByRole("button");
+  it("shows proper heading when rendered", () => {
+    const {getByRole} = render(Button, { props });
+    const button = getByRole("button");
 
-    expect(() => button).not.toThrow();
-    expect(button.classList).not.toBe("");
     expect(button.classList).toContain("custom-class");
     expect(button.getAttribute("type")).toEqual("submit");
     expect(button.getAttribute("name")).toEqual(props.name);
     expect(button.getAttribute("style")).toEqual(props.style);
   });
+
+  it("test on click event", () => {
+    const {component, getByRole} = render(Button, { props });
+    //mock function
+    const mock = jest.fn();
+    const button = getByRole("button");
+    //set event
+    component.$on('click', mock);
+
+    //click button
+    fireEvent.click(button);
```

Comment:
line 42 is not async since it's just setting the click event for the function, i was following the example in [here](https://sveltesociety.dev/recipes/testing-and-debugging/unit-testing-svelte-component)

i cannot find fireEvent.click being used that way, so i didn't use await

Reply:
i think better if use await, fixed